catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/RefreshRegistryEntries.java

```diff
@@ -292,19 +358,73 @@ public void setEnableDelete(boolean enableDelete) {
         this.enableDelete = enableDelete;
     }
 
+    public void setExecutor(ScheduledExecutorService executor) {
+        this.executor = executor;
+    }
+
+    public void setRefreshIntervalSeconds(int refreshIntervalSeconds) {
+        if (this.refreshIntervalSeconds == refreshIntervalSeconds) {
+            return;
+        }
+        this.refreshIntervalSeconds = refreshIntervalSeconds;
+        if (scheduledTask != null) {
+            this.scheduledTask.cancel(false);
+            this.init();
+        }
+    }
+
+    public void setTaskWaitTimeSeconds(Integer taskWaitTimeSeconds) {
+        this.taskWaitTimeSeconds = taskWaitTimeSeconds;
+    }
+
+    public void destroy() {
+        executor.shutdown();
+        try {
+            if (!executor.awaitTermination(SHUTDOWN_TIMEOUT_SECONDS, TimeUnit.SECONDS)) {
+                executor.shutdownNow();
+                if (!executor.awaitTermination(SHUTDOWN_TIMEOUT_SECONDS, TimeUnit.SECONDS)) {
+                    LOGGER.error("Thread pool didn't terminate");
+                }
+            }
+        } catch (InterruptedException e) {
+            executor.shutdownNow();
+        }
+    }
+
+    static class RemoteResult {
+
+        private String registryId;
+
+        private Map<String, Metacard> remoteRegistryMetacards;
+
+        public RemoteResult(String registryId, Map<String, Metacard> remoteRegistryMetacards) {
+            this.registryId = registryId;
+            this.remoteRegistryMetacards = remoteRegistryMetacards;
+        }
+
+        public String getRegistryId() {
+            return registryId;
+        }
+
+        public Map<String, Metacard> getRemoteRegistryMetacards() {
+            return remoteRegistryMetacards;
+        }
+
+    }
+
     private static class RemoteRegistryResults {
 
-        Map<String, Metacard> remoteRegistryMetacards;
+        private Map<String, Metacard> remoteRegistryMetacards;
 
-        List<String> registryStoresQueried;
+        private List<String> registryStoresQueried;
 
-        List<String> failureList;
+        private List<String> failureList;
 
         public RemoteRegistryResults(Map<String, Metacard> remoteRegistryMetacards,
                 List<String> failureList, List<String> storesQueried) {
-            this.remoteRegistryMetacards = remoteRegistryMetacards;
-            this.failureList = failureList;
-            this.registryStoresQueried = storesQueried;
+            this.remoteRegistryMetacards = Collections.unmodifiableMap(remoteRegistryMetacards);
+            this.failureList = Collections.unmodifiableList(failureList);
```

Comment:
Would it be better to return a new map and new list instead of an unmodifiableMap and unmodifiableList in the getter methods?